apps/codexgraph_agent/pages/components/page.py

```diff
@@ -0,0 +1,351 @@
+import os
+import time
+from abc import ABC, abstractmethod
+from datetime import datetime
+
+import json
+import streamlit as st
+from apps.codexgraph_agent.pages.components.setting import setting
+from apps.codexgraph_agent.pages.components.sidebar import sidebar
+from apps.codexgraph_agent.pages.components.states import (
+    get_json_files, initialize_page_state)
+from modelscope_agent.environment.graph_database import GraphDatabaseHandler
+from modelscope_agent.environment.graph_database.build import \
+    build_graph_database
+
+
+def get_llm_config(llm_name):
+
+    if llm_name == 'deepseek-coder':
+
+        llm_config = {
+            'model': 'deepseek-coder',
+            'api_base': 'https://api.deepseek.com',
+            'model_server': 'openai'
+        }
+
+    elif llm_name == 'gpt-4o':
+
+        llm_config = {
+            'model': 'gpt-4o-2024-05-13',
+            'api_base': 'https://api.openai.com/v1',
+            'model_server': 'openai'
+        }
+
+    else:
+        return None
+
+    return llm_config
+
+
+def agent_test_run(user_query, file_path, call_back):
+    for i in range(10):
+        message = f'Processing {i + 1}/{10}...'
+        call_back(message, 'assistant', '🤖')
+
+        time.sleep(1)  # Simulate work by sleeping for 1 second
+    return 'test'
+
+
+def update_progress_bar(page_name, progress):
+    st.session_state[page_name]['progress_bar'].progress(int(progress * 100))
+
+
+class PageBase(ABC):
+
```

Comment:
add a simple comment to introduce the functionality of this class

Reply:
ok

Reply:
Comment of `PageBase` has been added to the code.